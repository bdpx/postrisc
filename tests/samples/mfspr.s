.text
    alloc	61
    write	"\ntest write: special register"
    write	"ip      %s(ip)"
    write	"eip     %s(eip)"
    write	"eca     %s(eca)"
    write	"fpcr    %s(fpcr)"
    write	"rsc     %s(rsc)"
    write	"rsp     %s(rsp)"
    write	"bsp     %s(bsp)"
    write	"peb     %s(peb)"
    write	"teb     %s(teb)"
    write	"itc     %s(itc)"
    write	"itm     %s(itm)"
    write	"psr     %s(psr)"
    write	"pta     %s(pta)"
    write	"iva     %s(iva)"
    write	"kip     %s(kip)"
    write	"ksp     %s(ksp)"
    write	"krsp    %s(krsp)"
    write	"iip     %s(iip)"
    write	"iipa    %s(iipa)"
    write	"ipsr    %s(ipsr)"
    write	"cause   %s(cause)"
    write	"ifa     %s(ifa)"
    write	"iib     %s(iib)"
    write	"tpr     %s(tpr)"
    write	"lid     %s(lid)"
    write	"irr0    %s(irr0)"
    write	"irr1    %s(irr1)"
    write	"irr2    %s(irr2)"
    write	"irr3    %s(irr3)"
    write	"isr0    %s(isr0)"
    write	"isr1    %s(isr1)"
    write	"isr2    %s(isr2)"
    write	"isr3    %s(isr3)"
    write	"tsv     %s(tsv)"
    write	"cmcv    %s(cmcv)"
    write	"pmv     %s(pmv)"

    write	"\ntest mfspr: read special register"

    mf.spr	%r12, %ip
    write	"ip      %x64(r12)"

    mf.spr	%r12, %eip
    write	"eip     %x64(r12)"

    mf.spr	%r12, %eca
    write	"%x64(r12)"

    mf.spr	%r12, %fpcr
    write	"%x64(r12)"

    mf.spr	%r12, %rsc
    write	"%x64(r12)"

    mf.spr	%r12, %rsp
    write	"%x64(r12)"

    mf.spr	%r12, %bsp
    write	"%x64(r12)"

    mf.spr	%r12, %peb
    write	"%x64(r12)"

    mf.spr	%r12, %teb
    write	"%x64(r12)"

    mf.spr	%r12, %itc
    write	"%x64(r12)"

    mf.spr	%r12, %itm
    write	"%x64(r12)"

    mf.spr	%r12, %psr
    write	"%x64(r12)"

    mf.spr	%r12, %pta
    write	"%x64(r12)"

    mf.spr	%r12, %iva
    write	"%x64(r12)"

    mf.spr	%r12, %kip
    write	"%x64(r12)"

    mf.spr	%r12, %ksp
    write	"%x64(r12)"

    mf.spr	%r12, %krsp
    write	"krsp    %x64(r12)"

    mf.spr	%r12, %iip
    write	"iip     %x64(r12)"

    mf.spr	%r12, %iipa
    write	"iipa    %x64(r12)"

    mf.spr	%r12, %ipsr
    write	"ipsr    %x64(r12)"

    mf.spr	%r12, %cause
    write	"cause   %x64(r12)"

    write	"%s(ifa)"
    mf.spr	%r12, %ifa
    write	"ifa     %x64(r12)"

    mf.spr	%r12, %iib
    write	"iib     %x128(r12)"

    mf.spr	%r12, %tpr
    write	"tpr     %x64(r12)"

    mf.spr	%r12, %lid
    write	"lid     %x64(r12)"

    mf.spr	%r12, %irr0
    write	"irr0    %x64(r12)"

    mf.spr	%r12, %irr1
    write	"irr1    %x64(r12)"

    mf.spr	%r12, %irr2
    write	"irr2    %x64(r12)"

    mf.spr	%r12, %irr3
    write	"irr3    %x64(r12)"

    mf.spr	%r12, %isr0
    write	"%x64(r12)"

    mf.spr	%r12, %isr1
    write	"%x64(r12)"

    mf.spr	%r12, %isr2
    write	"%x64(r12)"

    mf.spr	%r12, %isr3
    write	"%x64(r12)"

    mf.spr	%r12, %tsv
    write	"%x64(r12)"

    mf.spr	%r12, %cmcv
    write	"%x64(r12)"

    mf.spr	%r12, %pmv
    write	"%x64(r12)"

    write	"end test mfspr"
.end
