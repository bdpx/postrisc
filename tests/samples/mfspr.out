
test write: special register
ip      fffffffff1230000
eip     fffffffff1230280
eca     0000000000000000
fpcr    0000000000000000 fe=vzoux rm=0
rsc     000000000000003d sof=61 bof=0 soc=0 ndirty=0 bod=0
rsp     fffffffff1730000
bsp     fffffffff1630000
peb     0000000000000000
teb     0000000000000000
itc     0000000000000004
itm     0000000000000000
psr     0000000000020000 pm=0 id=0 dd=0 lp=0 tb=0 ss=0 ic=0 ib=0 pl=0 pp=0 ri=2 future=00000000000000000000000000000000
pta     00000012345001c7 v=1 pta=0000001234500000 pageoffset=14 nlevels=3
iva     0000000000000000
kip     ffffffff00000000
ksp     0000000000000000
krsp    0000000000000000
iip     fffffffff1230000
iipa    fffffffff1230060
ipsr    0000000000000000 pm=0 id=0 dd=0 lp=0 tb=0 ss=0 ic=0 ib=0 pl=0 pp=0 ri=0 future=00000000000000000000000000000000
cause   0000000000000000 x=0 w=0 r=0 a=0 n=0 d=0 ei=0 code=0 vector=0
ifa     0000000000000000
iib     0000000000000000_0000000000000000
tpr     00000000000000ff vector=255
lid     0000000000000000
irr0    0000000000000000
irr1    0000000000000000
irr2    0000000000000000
irr3    0000000000000000
isr0    0000000000000000
isr1    0000000000000000
isr2    0000000000000000
isr3    0000000000000000
tsv     0000000000008000 vector=0 mask=1
cmcv    0000000000008000 vector=0 mask=1
pmv     0000000000008000 vector=0 mask=1

test mfspr: read special register
ip      fffffffff12300d0
eip     fffffffff1230280
0000000000000000
0000000000000000
000000000000003d
fffffffff1730000
fffffffff1630000
0000000000000000
0000000000000000
0000000000000014
0000000000000000
0000000000020000
00000012345001c7
0000000000000000
ffffffff00000000
0000000000000000
krsp    0000000000000000
iip     fffffffff1230000
iipa    fffffffff1230190
ipsr    0000000000000000
cause   0000000000000000
0000000000000000
ifa     0000000000000000
iib     0000000000000000_0000000000000000
tpr     00000000000000ff
lid     0000000000000000
irr0    0000000000000000
irr1    0000000000000000
irr2    0000000000000000
irr3    0000000000000000
0000000000000000
0000000000000000
0000000000000000
0000000000000000
0000000000008000
0000000000008000
0000000000008000
end test mfspr
